### DISCUSSION TITLE
nest form, any other good idea?

### DISCUSSION THREAD

**Discussion by leosxie** [react-hook-form/react-hook-form#4009] [Q&A]
online: https://codesandbox.io/s/react-hook-form-demos-forked-i4z9o?file=/src/index.tsx
online demo2: https://codesandbox.io/s/react-hook-form-v6-validationschema-forked-zuih2?file=/src/index.js
in parent form set subform value, subform field value change but not change in UI 
![image](https://user-images.githubusercontent.com/181152/105632482-3f949080-5e8e-11eb-9a8c-7b65c281d66d.png)

I recently encountered a need for a giant form with a lot of nesting and data management.
The nested form is an independent component and will be used elsewhere. My current hope is that all data verification changes are performed at the outermost periphery. In the process of using useForm, it is found that there is some problem in changing the value of the nested form outside. The value can be changed correctly but not displayed correctly

**Comment by bluebill1049** [comment]
can you share some steps to reproduce the issue?

**Reply by leosxie** [reply]
@bluebill1049  tks your reply. demo2, on parent form change first name , when firstname.length > 3 ,will change subForm age value. but ui not change. i just want find a good way to manager nests form  aform->bform->cform->dform，but every form is independent, when form combination，aform can change other level form filed values.

**Reply by bluebill1049** [reply]
each instance shares its own resource, that's not going to work. options

1. use single useForm
2. use a state management lib to store those values and get them updated.

**Reply by leosxie** [reply]
@bluebill1049   for example:  a big container has  aform->bform->cform->dform .but cform maybe use in otherplace.
if use store to management state, its mean subform not independent。
